main.cpp

```diff
@@ -49,14 +49,15 @@ Project 4: Part 4 / 9
 #include "IntType.h"
 #include "FloatType.h"
 #include "DoubleType.h"
+#include <iostream>
 
 
 struct Point
 {
-    Point(DoubleType& x_, DoubleType& y_);
-    Point(IntType& x_, IntType& y_);
-    Point(FloatType& x_, FloatType& y_);
-    Point(float x_, float y_) : x { x_} , y { y_} {}
+    Point(const DoubleType& x_, const DoubleType& y_) : x {static_cast<float>(x_)}, y {static_cast<float>(y_)}{}
+    Point(const IntType& x_, const IntType& y_) :       x {static_cast<float>(x_)}, y {static_cast<float>(y_)}{}
+    Point(const FloatType& x_, const FloatType& y_) :   x {static_cast<float>(x_)}, y {static_cast<float>(y_)}{}
+    
```

Comment:
Learn to use forwarding constructors, that forward arguments to the primary constructor, which you seem to have deleted.
```
Point(const DoubleType& x_, const DoubleType& y_) : Point(static_cast<float>(x_), static_cast<float>(y_) { }
```
- [ ] revise